backend/adapter_utils.go

```diff
@@ -1,173 +0,0 @@
-package backend
-
-import (
-	"context"
-	"errors"
-	"fmt"
-	"net/http"
-	"sync"
-	"time"
-
-	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
-	"github.com/grafana/grafana-plugin-sdk-go/backend/tracing"
-	"github.com/prometheus/client_golang/prometheus"
-	"go.opentelemetry.io/otel/attribute"
-	"go.opentelemetry.io/otel/trace"
-)
-
-type handlerWrapperFunc func(ctx context.Context) (RequestStatus, error)
-
-func setupContext(ctx context.Context, endpoint Endpoint) context.Context {
-	ctx = WithEndpoint(ctx, endpoint)
-	ctx = propagateTenantIDIfPresent(ctx)
-
-	return ctx
-}
-
-func wrapHandler(ctx context.Context, pluginCtx PluginContext, next handlerWrapperFunc) error {
-	ctx = setupHandlerContext(ctx, pluginCtx)
-	wrapper := errorWrapper(next)
-	wrapper = logWrapper(wrapper)
-	wrapper = metricWrapper(wrapper)
-	wrapper = tracingWrapper(wrapper)
-	_, err := wrapper(ctx)
-	return err
-}
-
-func setupHandlerContext(ctx context.Context, pluginCtx PluginContext) context.Context {
-	ctx = initErrorSource(ctx)
-	ctx = WithGrafanaConfig(ctx, pluginCtx.GrafanaConfig)
-	ctx = WithPluginContext(ctx, pluginCtx)
-	ctx = WithUser(ctx, pluginCtx.User)
-	ctx = withContextualLogAttributes(ctx, pluginCtx)
-	ctx = WithUserAgent(ctx, pluginCtx.UserAgent)
-	return ctx
-}
-
-func errorWrapper(next handlerWrapperFunc) handlerWrapperFunc {
-	return func(ctx context.Context) (RequestStatus, error) {
-		status, err := next(ctx)
-
-		if err != nil && IsDownstreamError(err) {
-			if innerErr := WithDownstreamErrorSource(ctx); innerErr != nil {
-				return RequestStatusError, fmt.Errorf("failed to set downstream status source: %w", errors.Join(innerErr, err))
-			}
-		}
-
-		return status, err
-	}
-}
-
-var pluginRequestCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
-	Namespace: "grafana_plugin",
-	Name:      "request_total",
-	Help:      "The total amount of plugin requests",
-}, []string{"endpoint", "status", "status_source"})
-
-var once = sync.Once{}
-
-func metricWrapper(next handlerWrapperFunc) handlerWrapperFunc {
```

Comment:
Yes it was added in https://github.com/grafana/grafana-plugin-sdk-go/pull/1117 as explained in the description. It was added so Grafana could start use it sooner rather than later given this is quite a lot of changes